File: Data Structures & Algorithms/minimum-stack/submission-5.cs
public class MinStack
{
    Stack<int> stack = new Stack<int>();
    Stack<int> minStack = new Stack<int>();

    public MinStack()
    {
    }

    public void Push(int val)
    {
        stack.Push(val);
        minStack.Push(Math.Min(minStack.Count == 0 ? val : minStack.Peek(), val));
    }

    public void Pop()
    {
        if (stack.Count == 0)
        {
            return;
        }

        stack.Pop();
        minStack.Pop();
    }

    public int Top()
    {
        return stack.Peek();
    }

    public int GetMin()
    {
        return minStack.Peek();
    }
}

File: Data Structures & Algorithms/crawler-log-folder/submission-5.cpp

class Solution {
public:
    int minOperations(vector<string> &logs) {
        stack<string> s;
        for (string l: logs) {
            if (l == "../") {
                if (s.size() > 0) s.pop();
            }
            else if (l == "./") continue;
            else {
                s.push(l);
            }
        }

        return s.size();
    }
};